mmengine/model/fast_conv_bn_eval.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from operator import attrgetter
+from typing import List, Tuple, Union
+
+import torch
+import torch.fx as fx
+import torch.nn as nn
+from mmcv.cnn import ConvModule
+
+if not hasattr(ConvModule, 'turn_on_fast_conv_bn_eval'):
+    raise RuntimeError(
+        'The MMCV you use does not support "fast_conv_bn_eval" feature.'
+        'Please install the latest version of MMCV to use this feature.')
+
+
+# Helper function to split a qualname into parent path and last atom.
+def _parent_name(target: str) -> Tuple[str, str]:
+    """Splits a qualname into parent path and last atom.
+
+    For example, `foo.bar.baz` -> (`foo.bar`, `baz`)
+    """
+    *parent, name = target.rsplit('.', 1)
+    return parent[0] if parent else '', name
+
+
+def replace_sub_module(model, name, new_module):
+    # Remove the original module from the model
+    # usage: replace_sub_module(model, 'layer1.block2.conv2', conv)
+    parent_name, name = _parent_name(name)
+    if parent_name != '':
+        getter = attrgetter(parent_name)
+        parent = getter(model)
```

Comment:
Can we directly use `getattr` rather than the two lines?

Reply:
Fixed in [0de92fa](https://github.com/open-mmlab/mmengine/pull/1202/commits/0de92fa65e066592ace48491a31a94d4936d783d). Note that `getattr` does not work here. The `parent_name` might have multiple levels like `layer1.bottleneck`. Therefore, we need  `attrgetter`.